src/lib.rs

```diff
@@ -130,6 +131,12 @@ pub enum Expr {
     Shl(Box<Expr>, Box<Expr>),
     Add(Box<Expr>, Box<Expr>),
     Sub(Box<Expr>, Box<Expr>),
+
+    FromU16Be(Box<Expr>),
+    FromU16Le(Box<Expr>),
+    FromU32Be(Box<Expr>),
+    FromU32Le(Box<Expr>),
```

Comment:
The names of these are a bit funky aren't they, because `FromU32Be` means "applies to a tuple of four U8 values and returns a single U32 value by interpreting them in big endian order", it feels like the "FromU32" part of the name is almost a misdirection given that it's also "From4U8" and "ToU32"!

Reply:
What about `U32FromBe`?

Reply:
I can do that for now! Though I feel like eventually it would be nice to have a name distinct from the format name.